src/main/java/net/sf/jabref/shared/DBMSConnectionProperties.java

```diff
@@ -75,4 +91,45 @@ public void setPassword(String password) {
         this.password = password;
     }
 
+    /**
+     *  Gets all required data from {@link SharedDatabasePreferences} and sets them if present.
+     */
+    private void setFromPreferences(SharedDatabasePreferences prefs) {
+        if (prefs.getType().isPresent()) {
+            Optional<DBMSType> dbmsType = DBMSType.fromString(prefs.getType().get());
+            if (dbmsType.isPresent()) {
+                this.type = dbmsType.get();
+            }
+        }
+
+        if (prefs.getHost().isPresent()) {
```

Comment:
you could use ``prefs.getHost().ifPresent(prefsHost -> this.host = prefsHost)``.
This avoids the if statements